source/common/event/win32/signal_impl.cc

```diff
@@ -0,0 +1,45 @@
+#include "common/api/os_sys_calls_impl.h"
+#include "common/event/dispatcher_impl.h"
+#include "common/event/signal_impl.h"
+
+#include "event2/event.h"
+
+namespace Envoy {
+namespace Event {
+
+SignalEventImpl::SignalEventImpl(DispatcherImpl& dispatcher, signal_t signal_num, SignalCb cb)
+    : cb_(cb) {
+
+  if (signal_num > eventBridgeHandlersSingleton::get().size()) {
+    ENVOY_BUG(false, "Attempting to create SignalEventImpl with a signal id that exceeds the "
```

Comment:
Should this be a PANIC/RELEASE_ASSERT? Isn't this a strict programming error?

Reply:
My train of though here is the following:

Assume that this is actually a bug that is introduced and this `ENVOY_BUG` would hit. As a user of Envoy do I prefer to have reduced signal handling i.e. I can not close envoy from `ctrl+c` or do I want Envoy to fail to start. 

If I were the user I would just prefer the former because I can still work around it and have Envoy working in my server. I would have a bad experience for sure but I wouldn't be blocked.

Again no strong opinion and happy to change it to `PANIC` just stating were I am coming from.

Reply:
It does seem like a programming error.

Reply:
Fair. Changed it into a panic in this iteration.